### DISCUSSION TITLE
Notation rc.1 release check-in since 11/17/2022

### DISCUSSION THREAD

**Discussion by yizha1** [notaryproject/notaryproject#205] [General]
Notation rc.1 status on 11/17/2022, check `comments` column of [rc.1-notdone](https://github.com/orgs/notaryproject/projects/10/views/8) for details.

- Unblock Ratify, **done** (verified by ratify team)
   - https://github.com/notaryproject/notation-go/issues/143
		
- Notation-go refactoring, **2 days delay**, 2 PRs left, **request re-review**
  - Sign interface: https://github.com/notaryproject/notation-go/pull/200
  - https://github.com/notaryproject/notation-go/pull/207
  
- Notation cli update to notation-go interface, **on track**
  -  https://github.com/notaryproject/notation/pull/445
		
- Notation using OCI artifact manifest to store signatures, **on track**
  - Notation-go is done
  - Notation cli ongoing, https://github.com/notaryproject/notation/pull/445
		
- Using timeout as the restricting mechanism fo signature verification, **risky**
  - https://github.com/notaryproject/notation-go/issues/201
  - Can we support "Max Signature Attempts" first for rc.1? We also need to discuss where to put the configuration (new flag for verify command, or new configuration parameter or both)
	
- Tag to digest UX improvements, **risky**
   - https://github.com/notaryproject/roadmap/issues/61
   - Spec is still under PR review (cover part of debug options)
   - Implementation pending

- Debug options, **risky**
   - https://github.com/notaryproject/notation/issues/300
   - Partly covered by "Tag to digest UX improvements"
   - **Need align the next step.** Maybe this issue can be shared by two teams, since it is just to add debug info in the code.
	
- E2E testing (**suggest removing from rc.1**)
  - https://github.com/notaryproject/roadmap/issues/47

- Documentation:
  - Trust policy document **(Toddy could you share the right link)**
  - Other documentations are less important for rc.1

**Comment by vaninrao10** [comment]
1. Tag to digest UX improvements, risky
     - https://github.com/notaryproject/roadmap/issues/61 - We can complete the Spec and get the PR to closure and move the    
        implementation to POST rc.1
2. Add Expiry in generate envelop to be passed to plugin.
     - https://github.com/notaryproject/notation/issues/443 - Should be part of rc.1
3. Certificate Revocation
     - https://github.com/notaryproject/notaryproject/issues/72  - Should be part of rc.1
4. Using timeout as the restricting mechanism for signature verification, 
     - https://github.com/notaryproject/notation-go/issues/201 - Updated the issue as per the discussion. Define the CLI behavior and 
        defaults in RC1.
5. E2E testing (suggest moving to post rc.1)
      -  https://github.com/notaryproject/roadmap/issues/47
6. TODOs in signature Package
      -   https://github.com/notaryproject/notation-go/issues/80 - This is the PR review stage and hence can probably make it to rc.1
7. Deadcode clean up
      -  https://github.com/notaryproject/notation/issues/392 - This can be moved to post rc.1
8. Debug options
      -  Can be moved to Post rc.1 if no concerns.
9. We also need to talk about the fall back logic to be in the ORAS Lib or Notation Client. - Still not discussed ? and this should be part of rc.1 ?

**Comment by yizha1** [comment]
@priteshbandi any update on this issue? https://github.com/notaryproject/notation/issues/443. The spec need to be updated by you.

@vaninrao10 per last discussion, fallback to image manifest will be the scope of rc.2. We can confirm this again on Nov 21 community call

**Comment by yizha1** [comment]
**Notation CLI**
- https://github.com/notaryproject/notation/pull/445, **+1 by one org**
- https://github.com/notaryproject/notation/pull/440, **request re-review**
- https://github.com/notaryproject/notation/pull/439, **request re-review**
- https://github.com/notaryproject/notation/pull/394, **ongoing**

**Notation-go**
- https://github.com/notaryproject/notation-go/pull/210, 
- https://github.com/notaryproject/notation-go/pull/208, **request re-review**
- https://github.com/notaryproject/notation-go/pull/207, 
- https://github.com/notaryproject/notation-go/pull/211, 
- https://github.com/notaryproject/notation-go/pull/178, **removed from rc.1 scope** 
- https://github.com/notaryproject/notation-go/pull/213, **request re-review**

**Notation-core-go**
- https://github.com/notaryproject/notation-core-go/pull/95, 

**notaryproject**
- https://github.com/notaryproject/notaryproject/pull/209, **done**
- https://github.com/notaryproject/notaryproject/pull/203, **done**
- https://github.com/notaryproject/notaryproject/pull/148, **done**

**Reply by FeynmanZhou** [reply]
For the CLI help doc in RC.1, apart from https://github.com/notaryproject/notation/pull/394, I will raise a new PR to update the command notation sign and verify later.

**Comment by yizha1** [comment]
Status on Nov 29. 

### Summary
It is proposed to postpone rc.1 release for a week. The new date is **Dec 5, 2022**

**PR Status**
- notation-core-go, done
- notatoryproject
   - [Pass expiry to generate-envelope plugin](https://github.com/notaryproject/notaryproject/pull/211), **==>ready to merge**
- notation-go
  - [update: bump up dependencies](https://github.com/notaryproject/notation-go/pull/219), ready to merge
- notation CLI
  - https://github.com/notaryproject/notation/pull/445, **==>approved by one org, under review**
  - verify spec:  https://github.com/notaryproject/notation/pull/440, **==>request re-review**
  - sign spec:  https://github.com/notaryproject/notation/pull/439, **==>request re-review**
  - CLI command help doc
    - https://github.com/notaryproject/notation/pull/454, **==>under review**
    - https://github.com/notaryproject/notation/pull/394, **==>under review**
				
**What's more** 

Add back work item: [debug logs for notation sign and verify commands](https://github.com/notaryproject/notation/issues/300), because it's hard to do troubleshooting for some verification failures, see results https://github.com/notaryproject/notation/discussions/452
> Maybe we cannot complete all the debug logs we want to add, but make sure we add necessary ones. We can complete all in rc.2 release.
